src/main/java/seedu/address/ui/cards/ApplicationCard.java

```diff
@@ -1,74 +1,174 @@
+import seedu.address.model.application.ApplicationItem;
 
 /**
  * todo Javadocs
  */
-public class ApplicationCard extends Card {
+public class ApplicationCard extends Card<ApplicationItem> {
 
     //Image Link
     private static final String CALENDAR_IMAGE_LINK = "/images/calendar.png";
 
+    //FXML properties
+    private static final String MIN_WIDTH_STATUS_BOX = " -fx-min-width: 60;";
+    private static final String MAX_WIDTH_STATUS_BOX = " -fx-max-width: 60;";
+    private static final String MIN_HEIGHT_STATUS_BOX = " -fx-min-height: 20;";
+    private static final String MAX_HEIGHT_STATUS_BOX = " -fx-max-height: 20;";
+    private static final String TRANSLATE_Y_STATUS_BOX = "-fx-translate-y: 5;";
+    private static final String ALIGNMENT_STATUS_BOX = "-fx-alignment: center;";
+    private static final String DEFAULT_STATUS_BOX = MIN_WIDTH_STATUS_BOX + MIN_HEIGHT_STATUS_BOX
+        + MAX_HEIGHT_STATUS_BOX + MAX_WIDTH_STATUS_BOX + TRANSLATE_Y_STATUS_BOX + ALIGNMENT_STATUS_BOX;
+    private static final String ACCEPTED_COLOR = "-fx-background-color: #3ADB9D;";
+    private static final String WAITING_COLOR = "-fx-background-color: #F4D014;";
+    private static final String REJECTED_COLOR = "-fx-background-color: #F02E62;";
+    private static final String APPLIED_COLOR = "-fx-background-color: #465891;";
+    private static final String INTERVIEW_COLOR = "-fx-background-color: #f86a23;";
+    private static final String OFFERED_COLOR = "-fx-background-color: #0e2578;";
+    private static final int IMAGE_HEIGHT_WIDTH = 23;
+
     /**
      * todo Javadocs
      */
-    public ApplicationCard(Person person, int displayedIndex) {
-        super(person, displayedIndex);
+    public ApplicationCard(ApplicationItem applicationItem, int displayedIndex) {
+        super(applicationItem, displayedIndex);
         initializeInternshipCardGui();
     }
 
     /**
      * todo Javadocs
      */
     private void initializeInternshipCardGui() {
-        // to be edited in the future!
-        initializeHeader(displayedIndex);
-        initializeBody();
-        initializeTags();
-        initializeDate();
-        initializeStatus();
+        setId(displayedIndex);
+        setName();
+        setTags();
+        setStatus();
+        setPeriod();
+        setWage();
+        setCompanyName();
+        setDate();
     }
 
     /**
      * todo Javadocs
      */
-    private void initializeStatus() {
-        setRandomStatus(); // to be change in when other classes are ready.
+    private void setName() {
+        Object jobTitle = mapping.get(JOB_TITLE_DISPLAY_NAME);
+        setName(jobTitle.toString());
     }
 
     /**
      * todo Javadocs
      */
-    private void initializeDate() {
-        Image calendarIcon = new Image(this.getClass().getResourceAsStream(CALENDAR_IMAGE_LINK));
-        calendar.setImage(calendarIcon);
-        date.setText("21/07/2020"); // to be changed later!
+    private void setTags() {
```

Comment:
Minor nit, but perhaps we could change this to `setRequirements` in the future? Just to keep all the namings consistent.

Reply:
yup I think I should change this to `setRequirements`